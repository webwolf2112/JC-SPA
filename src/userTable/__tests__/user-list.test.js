import React from 'react';
import { shallow } from 'enzyme';
import UserList from '../user-List';

describe( 'User Form Test ', () => {
    const deleteUserMock = jest.fn();
    const openFormModalMock = jest.fn();

    const defaultProps = {
        users: [{
            id: '1234',
            firstname: 'hello',
            lastname: 'world',
            userName: 'helloWorld123'
        }],
        deleteUser: deleteUserMock,
        updateUser: openFormModalMock
    };

    beforeEach( () => {
        jest.clearAllMocks();
    });

    it('Should render correct', () => {
        const wrapper = shallow(
          <UserList {...defaultProps} />,
        );
      
        expect(wrapper).toMatchSnapshot();
      });

      it('Should handle Delete', () => {
        const wrapper = shallow(
            <UserList {...defaultProps} />,
          );

          wrapper.find( 'button').first().simulate('click');
        
          expect( deleteUserMock ).toHaveBeenCalledWith( defaultProps.users[0].id );
      });

      it('Should call the open modal with the correct user index', () => {
        const wrapper = shallow(
            <UserList {...defaultProps} />,
          );

          wrapper.find( 'button').at(1).simulate('click');
        
          expect( openFormModalMock ).toHaveBeenCalledWith( 0 );
      });
});

