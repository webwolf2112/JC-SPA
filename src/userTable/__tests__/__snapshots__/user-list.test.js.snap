// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Form Test  Should render correct 1`] = `
<Fragment>
  <ul
    className="UserList"
  >
    <li>
      <span>
        First Name
      </span>
      <span>
        Last Name
      </span>
      <span>
        Email
      </span>
      <span />
    </li>
    <li
      key="1234"
    >
      <span>
        hello
      </span>
      <span>
        world
      </span>
      <span />
      <span>
        <button
          className="inverse"
          disabled={false}
          onClick={[Function]}
        >
          Delete User
        </button>
        <button
          onClick={[Function]}
        >
          Update User
        </button>
      </span>
    </li>
  </ul>
</Fragment>
`;
